ui/src/message_popup/message_popup_page.jsx

```diff
@@ -225,13 +308,23 @@ export default React.createClass({
           })}
         </RadioButtonGroup>
         <Divider />
+        <div>
+          <div style={styles.optionTitle}>Session Length: {this.state.sessionLength} {this.state.sessionLength===1 ? "question" : "questions"}</div>
+          <Slider key={competencyGroupValue} value={this.state.sessionLength} min={1} max={this.getQuestions().length} step={1} onChange={this.onSliderChange}/>
```

Comment:
Should this use `this.state.questions` instead?  Calling `getQuestions` isn't deterministic and may reshuffle the questions, and since `render` may be called multiple times it should just be a function of whatever the values in `props` and `state` are.